operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/DefaultEventSourceManager.java

```diff
@@ -30,23 +29,53 @@ public DefaultEventSourceManager(DefaultEventHandler defaultEventHandler, boolea
     }
   }
 
-  public void registerCustomResourceEventSource(
-      CustomResourceEventSource customResourceEventSource) {
-    this.customResourceEventSource = customResourceEventSource;
-    this.customResourceEventSource.addedToEventManager();
+  @Override
+  public void close() {
+    try {
+      lock.lock();
+      for (var entry : eventSources.entrySet()) {
+        try {
+          log.debug("Closing {} -> {}", entry.getKey(), entry.getValue());
+          entry.getValue().close();
+        } catch (Exception e) {
+          log.warn("Error closing {} -> {}", entry.getKey(), entry.getValue(), e);
+        }
+      }
+
+      eventSources.clear();
+    } finally {
+      lock.unlock();
+    }
   }
 
   @Override
-  public <T extends EventSource> void registerEventSource(String name, T eventSource) {
+  public final void registerEventSource(String name, EventSource eventSource) {
```

Comment:
Why did you make this method final?

Reply:
because it is invoked by the constructor so either the class is marked as final or this method need to be final to avoid issues in classes extending `DefaultEventSourceManager` and overriding `registerEventSource`

Reply:
Ah yes. I remember hearing about this but never ran into such issues in practice… Have you?

Reply:
yes, got bitten hard by it 😂 that's why I notice them